src/k8s-engine/kim.js

```diff
@@ -0,0 +1,173 @@
+'use strict';
+
+// This file should probably be an instance of ImageManager < EventEmitter.
+// We'll get there eventually
+
+const { EventEmitter } = require('events');
+const process = require('process');
+const { spawn } = require('child_process');
+const os = require('os');
+const fs = require('fs');
+const path = require('path');
+const util = require('util');
+const paths = require('xdg-app-paths')({ name: 'rancher-desktop' });
+const resources = require('../resources');
+const K8s = require('./k8s');
+
+const REFRESH_INTERVAL = 5 * 1000;
+
+async function runCommand(args) {
+  const child = spawn(resources.executable('kim'), args);
+  const result = { stdout: '', stderr: '' };
+
+  return await new Promise((resolve, reject) => {
+    child.stdout.on('data', (data) => {
+      result.stdout += data.toString();
+    });
+    child.stderr.on('data', (data) => {
+      result.stderr += data.toString();
+    });
+    child.on('exit', (code, sig) => {
+      if (code === 0) {
+        resolve(result);
+      } else if (sig !== undefined) {
+        reject({ ...result, signal: sig });
+      } else {
+        reject(result);
+      }
+    });
+  });
+}
+
+class Kim extends EventEmitter {
+  constructor() {
+    super();
+    this.notifiedMissingKim = false;
+    this.showedStderr = false;
+  }
+
+  start() {
+    fs.access(resources.executable('kim'),
```

Comment:
This doesn't make sense, if we're going to bundle kim in the installer package.

Reply:
noted